apps/namadillo/src/hooks/useTransactionWatcher.tsx

```diff
@@ -0,0 +1,31 @@
+import { useQuery } from "@tanstack/react-query";
+import { updateIbcTransactionStatus } from "atoms/integrations";
+import { myTransactionHistoryAtom } from "atoms/transactions/atoms";
+import { filterPendingTransactions } from "atoms/transactions/functions";
+import { useAtomValue } from "jotai";
+import { useMemo } from "react";
+import { IbcTransferTransactionData } from "types";
+import { useTransactionActions } from "./useTransactionActions";
+
+export const useTransactionWatcher = (): void => {
+  const { changeTransaction } = useTransactionActions();
+  const transactionHistory = useAtomValue(myTransactionHistoryAtom);
+  const pendingTransactions = useMemo(() => {
+    return transactionHistory.filter(filterPendingTransactions);
+  }, [transactionHistory]);
+
+  useQuery({
+    queryKey: ["transaction-status", pendingTransactions],
+    enabled: pendingTransactions.length > 0,
+    queryFn: async () => {
+      for (const tx of pendingTransactions) {
+        await updateIbcTransactionStatus(
+          tx.rpc,
+          tx as IbcTransferTransactionData,
+          changeTransaction
+        );
+      }
```

Comment:
we need to return something for `queryFn` other than `undefined`. Otherwise we got this error 

![Screenshot 2024-11-14 at 16 22 01](https://github.com/user-attachments/assets/3b158cff-ffdc-4ecb-baf5-bc26a86742f7)